monte_carlo.py

```diff
@@ -158,12 +159,18 @@ def transport_neutron(materials, bounds, tallies, fission_banks, first_round,
             neutron_distance -= tempd
 
             # add distance to cell flux
-            mesh.flux_add(cell, tempd)
+            mesh.flux_add(neutron.group, cell, tempd)
             
         # check interaction
         if neutron.alive:
             mat = mesh.get_material(neutron.get_cell())
-            neutron_interaction = sample_interaction(mat)
+
+            # samaple what group the neutron will scatter to
+            new_group = sample_scattered_group(group, mat.sigma_s)
```

Comment:
Only do this after we have determined the event was scattering. See the next comment.